File: appMain/src/main/java/com/cartlc/tracker/data/DataCollectionItem.java
package com.cartlc.tracker.data;

/**
 * Created by dug on 5/30/17.
 */

public class DataCollectionItem {
    public long    id; // row_id
    public long    collection_id; // project_id or collection_id
    public long    value_id;
    public int     server_id;
    public boolean isBootStrap;

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof DataCollectionItem) {
            return equals((DataCollectionItem) obj);
        }
        return super.equals(obj);
    }

    public boolean equals(DataCollectionItem other) {
        return collection_id == other.collection_id &&
                value_id == other.value_id;
    }

    public String toString() {
        StringBuilder sbuf = new StringBuilder();
        sbuf.append("[");
        sbuf.append(collection_id);
        sbuf.append(",");
        sbuf.append(value_id);
        sbuf.append(",");
        sbuf.append(server_id);
        sbuf.append("]");
        return sbuf.toString();
    }
}

File: appMain/src/main/java/com/cartlc/tracker/model/data/DataCollectionItem.kt
/**
 * Copyright 2018, FleetTLC. All rights reserved
 */
package com.cartlc.tracker.model.data

/**
 * Created by dug on 5/30/17.
 */

class DataCollectionItem {
    var id: Long = 0 // row_id
    var collection_id: Long = 0 // project_id or collection_id
    var value_id: Long = 0
    var server_id: Int = 0
    var isBootstrap: Boolean = false

    override fun equals(other: Any?): Boolean {
        return if (other is DataCollectionItem) {
            return collection_id == other.collection_id && value_id == other.value_id
        } else super.equals(other)
    }

    override fun toString(): String {
        val sbuf = StringBuilder()
        sbuf.append("[")
        sbuf.append(collection_id)
        sbuf.append(",")
        sbuf.append(value_id)
        sbuf.append(",")
        sbuf.append(server_id)
        sbuf.append("]")
        return sbuf.toString()
    }
}
